packages/fx-core/src/plugins/resource/bot/v2/plugin.ts

```diff
@@ -24,41 +25,43 @@ import * as utils from "../utils/common";
 import { HostTypeTriggerOptions } from "../question";
 import path from "path";
 import { AzureHostingFactory } from "../../../../common/azure-hosting/hostingFactory";
-import { isBotNotificationEnabled } from "../../../../common";
-import { AzureSolutionQuestionNames } from "../../../solution/fx-solution/question";
+import {
+  AzureSolutionQuestionNames,
+  BotOptionItem,
+  BotScenario,
+  MessageExtensionItem,
+} from "../../../solution/fx-solution/question";
 import {
   QuestionNames,
   TemplateProjectsConstants,
   TemplateProjectsScenarios,
   TriggerTemplateScenarioMappings,
 } from "../constants";
-import { PluginActRoles } from "../enums/pluginActRoles";
 import {
   BotTrigger,
   PluginBot,
-  QuestionBotScenarioToPluginActRoles,
   CommonStrings,
   Commands,
+  HostType,
+  HostTypes,
 } from "../resources/strings";
 import { CodeTemplateInfo } from "./interface/codeTemplateInfo";
 import { CommandExecutionError } from "../errors";
-import { BicepConfigs, HostType } from "../../../../common/azure-hosting/interfaces";
+import { BicepConfigs, ServiceType } from "../../../../common/azure-hosting/interfaces";
 import { mergeTemplates } from "../../../../common/azure-hosting/utils";
 import { getActivatedV2ResourcePlugins } from "../../../solution/fx-solution/ResourcePluginContainer";
 import { NamedArmResourcePluginAdaptor } from "../../../solution/fx-solution/v2/adaptor";
 import { ResourcePlugins } from "../../../../common/constants";
 
 export class TeamsBotV2Impl {
   async scaffoldSourceCode(ctx: Context, inputs: Inputs): Promise<Result<Void, FxError>> {
-    const workingPath = path.join(inputs.projectPath ?? "", "bot");
-    const hostTypeTriggers = inputs[QuestionNames.BOT_HOST_TYPE_TRIGGER];
-    let hostType;
-    if (Array.isArray(hostTypeTriggers)) {
-      const hostTypes = hostTypeTriggers.map(
-        (item) => HostTypeTriggerOptions.find((option) => option.id === item)?.hostType
-      );
-      hostType = hostTypes ? hostTypes[0] : undefined;
+    let workingPath: string;
```

Comment:
We can do a more refactor to remove duplicated logic of getting project path from inputs
```
let workingPath: string = inputs.projectPath ?? "";
if (inputs.platform !== Platform.VS) {
  workingPath = path.join(workingPath, "bot");
}
```